include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2508,7 +2522,7 @@ class binary_reader
             return false;
         }
 
-        if (input_format == input_format_t::bjdata && size_and_type.first != string_t::npos && size_and_type.first >= (1ull << (sizeof(std::size_t) * 8 - 1)))
+        if (input_format == input_format_t::bjdata && size_and_type.first != string_t::npos && size_and_type.second >= 256)
         {
             return false;
```

Comment:
This code path needs to emit a parse error.

I'll leave message and ID to you but this is what I'm using while testing.
```cpp
            auto last_token = get_token_string();
            return sax->parse_error(chars_read, last_token, parse_error::create(112, chars_read,
                                    exception_message(input_format, "unexpected ND array size information", "object"), nullptr));
```